Sources/NIOExtras/JSONRPCFraming+ContentLengthHeader.swift

```diff
@@ -0,0 +1,194 @@
+                    return .needMoreData
+                }
+
+                // Is this a header we actually care about?
+                if headerName == "content-length" {
+                    // Yes, let's parse the int.
+                    let headerValue = buffer.readString(length: newlineIndex - buffer.readableBytesView.startIndex + 1)!
+                    if let length = UInt32(headerValue.trimmed()) { // anything more than 4GB or negative doesn't make sense
+                        self.payloadLength = .init(length)
+                    } else {
+                        throw DecodingError.illegalContentLengthHeaderValue(headerValue)
+                    }
+                } else {
+                    // Nope, let's just skip over it
+                    buffer.moveReaderIndex(forwardBy: newlineIndex - buffer.readableBytesView.startIndex + 1)
+                }
+
+                // but in any case, we're now waiting for a new header or the end of the header block again.
+                self.state = .waitingForHeaderNameOrHeaderBlockEnd
+                return .continue
+            case .waitingForPayload(length: let length):
+                // That's the easiest case, let's just wait until we have enough data.
+                if let payload = buffer.readSlice(length: length) {
+                    // Cool, we got enough data, let's go back waiting for a new header block.
+                    self.state = .waitingForHeaderNameOrHeaderBlockEnd
+                    // And send what we found through the pipeline.
+                    context.fireChannelRead(self.wrapInboundOut(payload))
+                    return .continue
+                } else {
+                    return .needMoreData
+                }
+            }
+        }
+
+        /// Invoked when the `Channel` is being brough down.
+        public mutating func decodeLast(context: ChannelHandlerContext,
+                                        buffer: inout ByteBuffer,
+                                        seenEOF: Bool) throws -> DecodingState {
+            // Last chance to decode anything.
+            while try self.decode(context: context, buffer: &buffer) == .continue {}
+
+            if buffer.readableBytes > 0 {
+                // Oops, there are leftovers that don't form a full message, we could ignore those but it doesn't hurt to send
+                // an error.
+                throw ByteToMessageDecoderError.leftoverDataWhenDone(buffer)
+            }
+            return .needMoreData
+        }
+    }
+}
+
+extension String {
+    func trimmed() -> Substring {
+        var substring = self[...]
+        while substring.first.map({ $0.isNewline || $0.isWhitespace }) ?? false {
+            substring = substring.dropFirst()
+        }
+        while substring.last.map({ $0.isNewline || $0.isWhitespace }) ?? false {
+            substring = substring.dropLast()
+        }
+        return substring
+    }
```

Comment:
Feel free to ignore the code golf, but this seems like a verbose spelling for:

```swift
    func trimmed() -> Substring {
        guard let firstElementIndex = self.firstIndex(where: { !$0.isWhitespace }) else {
            return Substring("")
        }

        let lastElementIndex = self.reversed().first(where: { !$0.isWhitespace })!
        return self[firstElementIndex...lastElementIndex]
    }
```

Either way, you can remove `isNewline` because `isWhitespace` covers it.